### ISSUE TITLE
Bruop doesn't start on EKS cluster with IPv6 enabled.

### ISSUE THREAD

**Issue by pmacieje** [bottlerocket-os/bottlerocket-update-operator#178]
**Image I'm using:** bottlerocket-update-operator:v0.2.0

**Issue or Feature Request:** bruop-controller, apiservers and agent's doesn't start at all.

Errors:

**Controller:** "Error: ClientCreate { source: InferConfig(InferConfigError { in_cluster: ParseClusterUrl(InvalidUri(InvalidAuthority)), kubeconfig: ReadConfig(Os { code: 2, kind: NotFound, message: "No such file or directory" }, "/.kube/config") }) }"

**ApiServer:** {"v":0,"name":"apiserver","msg":"brupop apiserver failed.","level":50,"hostname":"brupop-apiserver-745c5cffd9-n7tdb","pid":1,"time":"2022-03-31T11:40:59.531118923+00:00","target":"apiserver","line":23,"file":"apiserver/src/main.rs","error":"Unable to create client: 'Failed to infer configuration: failed to infer config: in-cluster: (failed to parse cluster url: invalid authority), kubeconfig: (failed to read kubeconfig from '\"/.kube/config\"': No such file or directory (os error 2))'"}

**Additional info:** I've deployed bruop on EKS cluster v.1.21 with IPv6 enabled. Also deployed on IPv4 EKS cluster and there looks like work just fine.

**Comment by cbgbt** [comment]
Thanks for raising this issue. We'll take a look at Brupop in IPv6 clusters and see if we can reproduce.

**Comment by somnusfish** [comment]
I was able to reproduce this issue in the IPv6 cluster.

This issue is caused by `kube::client` inferring the default config from the cluster didn't format the IPv6 URI correctly. I have submitted a [bug report](https://github.com/kube-rs/kube-rs/pull/875) with a fix. Right now we are blocked on the upstream release.

To prevent this happen in future, we will add integration tests for IPv6 cluster (#186 ).
